Dockerfile

```diff
@@ -0,0 +1,30 @@
+ARG go_version=1.17
+
+# User
+FROM alpine:3.13.1 as user
+ARG uid=10001
+ARG gid=10001
+RUN echo "scratchuser:x:${uid}:${gid}::/home/scratchuser:/bin/sh" > /scratchpasswd
+
+# Certs
+FROM alpine:3.13.1 as certs
+RUN apk add -U --no-cache ca-certificates
+
+# Build
+FROM golang:${go_version} as build
+WORKDIR /code/
+ENV GO111MODULE=on
+ENV CGO_ENABLED=0
+COPY go.mod go.sum ./
+COPY ./cmd/ ./cmd/
+COPY ./internal/ ./internal/
+COPY ./pkg/ ./pkg/
+
+RUN go mod download
```

Comment:
It might be worth moving this to be directly after the copying of `go.mod` and `go.sum` but before the copying of the rest of the code so Docker's caching is utilised and the dependencies are only downloaded again if `go.mod` of `go.sum` change.

Reply:
Done